applications/fishing-map/src/features/map/map.selectors.ts

```diff
@@ -28,48 +39,34 @@ export const selectGlobalGeneratorsConfig = createSelector(
 )
 
 export const getGeneratorsConfig = createSelector(
-  [
-    selectWorkspaceDataviews,
-    selectDataviews,
-    selectFishingFilters,
-    selectDataviewsResourceQueries,
-    selectResources,
-  ],
-  (dataviews = [], urlDataviews, fishingFilters, resourceQueries, resources) => {
+  [selectWorkspaceDataviewsResolved, selectFishingFilters, selectResources],
+  (dataviews = [], fishingFilters, resources) => {
     // TODO add logic to merge 4Wings dataviews into one generator
-    return dataviews.map((dataview) => {
-      const urlDataview = urlDataviews?.find((urlDataview) => urlDataview.id === dataview.id)
-      const visible =
-        urlDataview?.config?.visible !== undefined ? urlDataview?.config?.visible : true
+    const generatorsConfig = dataviews.map((dataview) => {
       const filters = fishingFilters?.map((filter) => filter.id)
       const config = { ...dataview.config }
       if (config.type === Generators.Type.HeatmapAnimated && filters?.length) {
-        config.sublayers = (config.sublayers as HeatmapAnimatedGeneratorSublayer[])?.map(
-          (layer) => ({
-            ...layer,
-            filter: `flag=${filters.map((filter) => `'${filter}'`).join(',')}`,
-          })
-        )
+        config.sublayers = config.sublayers?.map((layer) => ({
+          ...layer,
+          filter: `flag=${filters.map((filter) => `'${filter}'`).join(',')}`,
+        }))
       }
 
       // Try to retrieve resource if it exists
       let data
-      const resourceQuery = resourceQueries.find((rq) => rq.dataviewId === dataview.id)
-      if (resourceQuery) {
-        const resource = resources[resourceQuery.url]
-        if (resource) {
-          data = resource.data
-        }
+      const { url } = resolveDataviewDatasetResource(dataview)
+      if (url && resources[url]) {
+        data = resources[url].data
       }
 
       return {
         ...config,
         // TODO Add vessel id for tracks ie
         // dataview.datasetsConfig[?].query.find(q => q.id === 'id').value
-        id: `${config.type}_${dataview.id}_SOME_UNIQUE_ID`,
-        visible,
+        id: `${config.type}_${dataview.configId}`,
```

Comment:
Actually, now that we have a proper uid generated (👍 ), I think this should simply be the dataviewInstance/dataviewConfig uid. 
Helps then to track stuff in style metadata.

Reply:
I agree but the `uid` is the [configId](https://github.com/GlobalFishingWatch/frontend/pull/118/files#diff-cdb6d9cab8c19208b3e9b07030066a29R85) now, see https://github.com/GlobalFishingWatch/frontend/pull/118/files#diff-975acf8e49c9a5d56577ed5bea82be8dR673.
It is mandatory when you add a dataview to a workspace instead of generated at runtime, or autogenerated when you add a new instance of the dataview to the workspace (for example, [adding a vessel from the search](https://github.com/GlobalFishingWatch/frontend/pull/118/files#diff-1d2e6e967661b4499f7c8472315e666aR54))